lib/new_relic/agent/configuration/default_source.rb

```diff
@@ -2570,6 +2570,84 @@ def self.enforce_fallback(allowed_values: nil, fallback: nil)
           :type => Integer,
           :allowed_from_server => false,
           :description => 'This value represents the total amount of memory available to the host (not the process), in mebibytes (1024 squared or 1,048,576 bytes).'
+        },
+        # security agent
+        :'security.agent.enabled' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => "If `true`, the security agent is loaded (a Ruby 'require' is performed)"
+        },
+        :'security.enabled' => {
+          :default => false,
+          :external => true,
+          :public => true,
+          :type => Boolean,
+          :allowed_from_server => false,
+          :description => 'If `true`, the security agent is started (the agent runs in its event loop)'
+        },
+        :'security.mode' => {
```

Comment:
Given the description text for `security.mode`, I think we should:

- Mark it as `public: false`
- Change the allowlist to %w[IAST]

If RASP support is added later, we can expose the config parameter for choosing it at that time.

Reply:
@fallwith #1 making `public: false` we can't do at the moment as we have mentioned all the security configs in documentation. #2 `%w[IAST]` Agree with this, update is on the way.

Reply:
Changes in place under this [commit](https://github.com/newrelic/newrelic-ruby-agent/pull/2752/commits/db297b85b582ce3fc273476bed28016634de1d85)

Reply:
@fallwith I need to add RASP again to allowlist reason being me and our QA guy use RASP for our testing purpose.